lib/poll.js

```diff
@@ -5,31 +5,31 @@
 // This file may not be copied, modified, or distributed
 // except according to those terms.
 
-'use strict';
+'use strict'
 
 var poll = {
-    pollPeriod: 2000,
-    commandStatus: function(client, commandId, deferred, initialTimeStamp, statusCallback) {
-         var intervalId = setInterval(function() {
-             client.commandsStatus(commandId)
-             .then( function(res) {
-                if(!res.body) {
-                  throw 'Incorrectly formatted response from commandsStatus -- response must have a body';
-                }
-                if(res.body.error) {
-                  deferred.reject({error: res.body.error});
-                  return;
-                }
-                var finalTimeStamp = Date.now();
-                if (res.body.state !== 'inProgress') {
-                  deferred.resolve({ body: res.body, response: res.response, timeElapsed: (finalTimeStamp - initialTimeStamp) });
-                  clearInterval(intervalId);
-                } else if (typeof statusCallback === 'function') {
-                  statusCallback({ body: res.body, response: res.response, error: res.error });
-                }
-             }, deferred.reject);
-        }, this.pollPeriod);
-    }
-};
+  pollPeriod: 2000,
+  commandStatus: function (client, commandId, deferred, initialTimeStamp, statusCallback) {
+    var intervalId = setInterval(function () {
+      client.commandsStatus(commandId)
+        .then(function (res) {
+          if (!res.body) {
+            throw new Error('Incorrectly formatted response from commandsStatus -- response must have a body')
+          }
+          if (res.body.error) {
+            deferred.reject({error: res.body.error})
```

Comment:
Don't we want to wrap this in an error?

Perhaps we should set this eslint rule - http://eslint.org/docs/rules/prefer-promise-reject-errors